src/util_cs/test_cs.rs

```diff
@@ -176,6 +176,17 @@ impl<Scalar: PrimeField> TestConstraintSystem<Scalar> {
         Default::default()
     }
 
+    pub fn scalar_inputs(&self) -> Vec<Scalar> {
+        self.inputs
+            .iter()
+            .map(|(scalar, _string)| *scalar)
+            .collect()
+    }
+
+    pub fn scalar_aux(&self) -> Vec<Scalar> {
+        self.aux.iter().map(|(scalar, _string)| *scalar).collect()
+    }
+
```

Comment:
This (and their alter ego in `WitnessCS`) seems orthogonal from the thrust of the PR, am I missing something?

Reply:
They are need for testing. For example, in the downstream [neptune PR](https://github.com/lurk-lab/neptune/pull/190):

https://github.com/lurk-lab/neptune/blob/3db8506f7ad551c683d2e20784240115ca723abe/src/circuit2_witness.rs#L444-L454